tests/unit/test_koji_wrapper.py

```diff
@@ -119,6 +123,89 @@ def test_returns_srpm_url(a_koji_wrapper, sample_build, sample_rpm_list):
     assert a_koji_wrapper.session.listRPMs.called
     assert isinstance(srpm_url,str)
 
+
+def test_build_srpm_url_none_srpm_and_build(a_koji_wrapper):
+    """
+    GIVEN we have a valid KojiWrapper with a session,
+    WHEN we call _build_srpm_url method with None
+    THEN we get back a TypeError
+    """
+
+    with pytest.raises(TypeError):
+        a_koji_wrapper._build_srpm_url(None, None)
+
+
+def test_build_srpm_url_none_build(a_koji_wrapper, sample_rpm_list):
+    """
+    GIVEN we have a valid KojiWrapper with a session,
+    WHEN we call _build_srpm_url method with sample_srpm_name and None
+    THEN we get back a TypeError
+    """
+
+    with pytest.raises(TypeError):
+        a_koji_wrapper._build_srpm_url(sample_srpm_name, None)
+
+
+def test_build_srpm_url_none_srpm(a_koji_wrapper, sample_build):
+    """
+    GIVEN we have a valid KojiWrapper with a session,
+    WHEN we call _build_srpm_url method with None and sample_build
+    THEN we get back a TypeError
+    """
+
+    with pytest.raises(TypeError):
+        a_koji_wrapper._build_srpm_url(None, sample_build)
+
+
+def test_build_srpm_url_empty_build_and_srpm_string(a_koji_wrapper):
+    """
+    GIVEN we have a valid KojiWrapper with a session,
+    WHEN we call _build_srpm_url method with empty strings
+    THEN we get back a TypeError
+    """
+
+    with pytest.raises(TypeError):
+        a_koji_wrapper._build_srpm_url('', '')
+
+
+def test_build_srpm_url_empty_build_string(a_koji_wrapper, sample_rpm_list):
+    """
+    GIVEN we have a valid KojiWrapper with a session,
+    WHEN we call _build_srpm_url method with sample_srpm_name and empty string
+    THEN we get back a TypeError
+    """
+
+    with pytest.raises(TypeError):
+        a_koji_wrapper._build_srpm_url(sample_srpm_name, '')
+
+
+def test_build_srpm_url_empty_srpm_string(a_koji_wrapper, sample_build):
+    """
+    GIVEN we have a valid KojiWrapper with a session,
+    WHEN we call _build_srpm_url method with empty string and sample_build
+    THEN we get back a TypeError
+    """
+
+    with pytest.raises(TypeError):
+        a_koji_wrapper._build_srpm_url('', sample_build)
+
+
+def test_build_srpm_url_positive(a_koji_wrapper, sample_build, sample_srpm):
```

Comment:
I like the thoroughness of your tests, but I think you should consider using parametrize decorator[1] here, as they are all the same test, but with different values.

[1] https://docs.pytest.org/en/latest/parametrize.html

Reply:
good idea, can this be a follow up refactor? or would you like to see it as a patch for this pull request before it's merged?